watcher.go

```diff
@@ -563,12 +567,11 @@ func (t *tracker) sweep(n Notifier) {
 	t.Lock()
 	defer t.Unlock()
 	used := make(map[string]struct{})
-	// remove tracked that were used
 	tmp := t.tracked[:0]
 	for _, tp := range t.tracked {
 		otherNotifier := tp.notify != n.ID()
 		if tp.inUse || otherNotifier {
-			tmp = append(tmp, tp.refresh())
```

Comment:
I'm not sure why a tracked pair is marked from being unused here (`refresh()`) meanwhile marking it as used by adding it to the `used` map. It gets subsequently deleted on the next `sweep` call.